uptane-standard.md

```diff
@@ -289,20 +289,309 @@ The Timestamp role SHALL produce and sign metadata indicating whether there are
+NumberOfKeys  = Length
+NumberOfRoles = Length
+RootMetadata  = NumberOfKeys PublicKeys NumberOfRoles TopLevelRoles
+
+; Adjust length to your needs.
+TopLevelRoles = 4(TopLevelRole)
+NumberOfURLs  = Length ; TAP 5: URLs pointing to the metadata file for this role.
+NumberOfKeyIds = Length
+TopLevelRole  = RoleType [NumberOfURLs] [URLs] NumberOfKeyIds Keyids Threshold
+```
+
 ### Targets Metadata {#targets_meta}
 
+At a minimum, a targets metadata file contains metadata (i.e., filename, hashes, length) about unencrypted images on a repository. The file MAY also contain two optional pieces of information: (1) custom metadata about which images should be installed by which ECUs, and whether encrypted images are available, and / or (2) other delegated targets roles that have been entrusted to sign images. This file is signed using the private keys of either the top-level targets role or a delegated targets role.
+
+The following example specifies all of the REQUIRED as well as all of the RECOMMENDED attributes for the body of targets metadata.
+```
+Number of Targets  = Natural ; Allowed to have no targets at all.
+TargetsMetadata    = NumberOfTargets Targets [TargetsDelegations] ; https://tools.ietf.org/html/rfc6025#section-2.4.2
+
+; Adjust length to your needs.
+Targets           = 1*128(TargetAndCustom)
+TargetAndCustom   = Target [Custom]
+NumberOfHashes    = Length
+Target            = URL Filename Length NumberOfHashes Hashes
+
+; The release counter is used to prevent rollback attacks on images when
+; only the director repository is compromised.
+; Every ECU should check that the release counter of its latest image is
+; greater than or equal to the release counter of its previous image.
+ReleaseCounter = Natural
+
+; The hardware identifier is used to prevent the director repository,
+; when it is compromised, from choosing images for an ECU that were not
+; meant for it.
+; Every ECU should check that the hardware ID of its latest image matches
+; its hardware ID.
+; An OEM MAY define other types of information to further restrict the
+; choices that can be made by a compromised director repository.
+HardwareIdentifier = Identifier
+
+; The ECU identifier specifies information, e.g., serial numbers, that the
+; director uses to point ECUs as to which images they should install.
+; Every ECU should check that the ECU ID of its latest image matches its
+; own ECU ID.
+EcuIdentifier = Identifier
+
+; This attribute MAY be used by the director to encrypt images per ECU.
+EncryptedTarget = Target
+
+; This attribute is used to specify additional information, such as which
+; images should be installed by which ECUs, and metadata about encrypted
+; images.
+; NOTE: The first 2 attributes are specified by both the image and
+; director repositories.
+; NOTE: The remaining attributes are specified only by the director
+; repository.
+Custom = [ReleaseCounter] [HardwareIdentifier] [EcuIdentifier] [EncryptedTarget] [EncryptedSymmetricKey]
```

Comment:
We might need more information here to indicate associated data like IV and nonce when encrypting images / targets